libtenzir/builtins/operators/python.cpp

```diff
@@ -319,13 +288,16 @@ class python_operator final : public crtp_operator<python_operator> {
                                codepipe.pipe().native_source(),
                                errpipe.pipe().native_sink()}},
         bp::detail::limit_handles_{}};
+      ::close(codepipe.pipe().native_source());
       if (code.empty()) {
         // The current implementation always expects a non-empty input.
         // Otherwise, it blocks forever on a `read` call.
         codepipe << " ";
       } else {
         codepipe << code;
       }
+      codepipe.flush();
+      ::close(codepipe.pipe().native_sink());
       codepipe.close();
       ::close(errpipe.pipe().native_sink());
```

Comment:
Why do this twice?

Reply:
I added a comment to explain why we call 2 close functions.